core/src/test/java/com/linecorp/armeria/internal/common/metric/NettyMetricsSupportTest.java

```diff
@@ -0,0 +1,63 @@
+package com.linecorp.armeria.internal.common.metric;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.linecorp.armeria.common.metric.MeterIdPrefix;
+import com.linecorp.armeria.common.metric.MoreMeters;
+import com.linecorp.armeria.common.metric.PrometheusMeterRegistries;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.netty.channel.DefaultEventLoopGroup;
+import io.netty.channel.EventLoopGroup;
+import org.junit.Test;
+
+import java.util.concurrent.CountDownLatch;
+
+public class NettyMetricsSupportTest {
+
+    private class BlockMe extends CountDownLatch implements Runnable {
+
+        BlockMe() {
+            super(1);
+        }
+
+        @Override
+        public void run() {
+            try { await(); } catch(Throwable ignored) { }
+        }
+    }
+
+
+    @Test
+    public void test() {
+        MeterRegistry registry = PrometheusMeterRegistries.newRegistry();
+        NettyMetricSupport.NettyMetrics metrics =
+                new NettyMetricSupport.NettyMetrics(
+                        registry,
+                        new MeterIdPrefix("foo")
+                );
+
+        BlockMe task = new BlockMe();
+
+        EventLoopGroup workers = new DefaultEventLoopGroup(2);
+        // Block both executors
+        workers.submit(task);
+        workers.submit(task);
+
+        workers.submit(() -> {});
+
+        metrics.add(workers);
+
+        // Check that API works as expected
+        assertThat(metrics.pendingTasks()).isEqualTo(1.0);
+        assertThat(metrics.numWorkers()).isEqualTo(2.0);
+
+        // Check that metrics are exported
+        assertThat(MoreMeters.measureAll(registry))
+            .containsEntry("foo.num.workers#value", 2.0)
+            .containsEntry("foo.pending.tasks#value", 1.0);
+
+        // Release & shutdown threads
+        task.countDown();
+        workers.shutdownGracefully();
```

Comment:
How about checking again if the number of pending tasks is zero? Could use `await()`:

```java
await().untilAsserted(() -> {
    assertThat(MoreMeters.measureAll(registry))
        .containsEntry("foo.num.workers#value", 2.0)
        .containsEntry("foo.pending.tasks#value", 0.0);
});
```